src/hotspot/share/opto/compile.cpp

```diff
@@ -3782,6 +3782,14 @@ void Compile::final_graph_reshaping_main_switch(Node* n, Final_Reshape_Counts& f
     }
     break;
   }
+#ifdef ASSERT
+  case Op_ConNKlass: {
+    const TypePtr* tp = n->as_Type()->type()->make_ptr();
+    ciKlass* klass = tp->is_klassptr()->exact_klass();
+    assert(!klass->is_interface() && !klass->is_abstract(), "Interface or abstract class pointers should not be compressed");
```

Comment:
Can you make this assert be instead:

#include "oops/compressedKlass.hpp"
...
assert(CompressedKlassPointers::is_encodable(klass), "should be encodable");

Reply:
Sure, good point. I updated the code.